### ISSUE TITLE
Redundancy of computing Loss in CustomTrainer - Bert

### ISSUE THREAD

**Issue by AmineOuerfellii** [emilienschultz/pyactivetigger#1030]
Hello 

while using `inputs.get("labels")` keeps labels in the dict  causing the model to compute its own loss internally and the custom compute_loss() computes again so this maybe a case of a wasted computation

after doing the change :

<img width="555" height="52" alt="Image" src="https://github.com/user-attachments/assets/5a3b4513-8e7d-43ee-9912-426d155c6b84" />

Worth noting too : 

from train_loader()
```
if loss == "cross_entropy":
    trainer = Trainer(
        model=bert_model,
        args=training_args,
        train_dataset=ds["train"],
        eval_dataset=eval_dataset,
        callbacks=[callback],
    )
```
although it's mentionned that loss is crossentropy but the loss computed internally is BCEwithlogits (ping me to add this to the PR draft ) 

Also and although it's a minor isssue but worth mentionning , the overwrite of the whole txt using dump might be exhausting , i mean in case if the file purpose is to serve as an info bus , i know this could lead to a little to probably a medium refactor of course based on the options we have (like making the server push updates, websocket...)

**Comment by emilienschultz** [comment]
Ok, it seems to be a real bug : the default Trainer branch is silently training with BCE, not CE

If you have some bandwidth and willing to fix it, it would be great !

For the 2 other points (the info bus and the wasted computation, you are right but it is minor). If you want to solve the wasted computation of loss, please do it in a separate PR.

Thanks for spotting this out !

**Comment by AmineOuerfellii** [comment]
Hey @emilienschultz  , thanks i'll clean all this out and ping you when it's done

**Comment by emilienschultz** [comment]
Hi @AmineOuerfellii just an update to know if you want to push a PR for this or if you prefer I do it ?

**Comment by AmineOuerfellii** [comment]
Hi @emilienschultz , i've already made the necessary fixes, you can check the PR i pushed

**Comment by emilienschultz** [comment]
Thanks @AmineOuerfellii for this PR !!

**Comment by AmineOuerfellii** [comment]
Your Welcome @emilienschultz
